cmd/cg-client/main.go

```diff
@@ -0,0 +1,342 @@
+	"net/url"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"gopkg.in/ini.v1"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	// Must be a global variable in the data segment so that the build
+	// process can inject the version number on the fly when building the
+	// binary. Use only from the Usage() function.
+	Version = "No version provided"
+)
+
+var (
+	certFilename         = flag.String("cert", filepath.Join(getUserHomeDir(), ".ssl", "keymaster.cert"), "A PEM eoncoded certificate file.")
+	keyFilename          = flag.String("key", filepath.Join(getUserHomeDir(), ".ssl", "keymaster.key"), "A PEM encoded private key file.")
+	baseURL              = flag.String("baseURL", "https://cloud-gate.symcpe.net", "location of the cloud-broker")
+	crededentialFilename = flag.String("credentialFile", filepath.Join(getUserHomeDir(), ".aws", "credentials"), "An Ini file with credentials")
+	askAdminRoles        = flag.Bool("askAdminRoles", false, "ask also for admin roles")
+	outputProfilePrefix  = flag.String("outputProfilePrefix", "saml-", "prefix to put to profile names $PREFIX$accountName-$roleName")
+	lowerCaseProfileName = flag.Bool("lowerCaseProfileName", true, "ask also for admin roles")
+	configFilename       = flag.String("configFile", filepath.Join(getUserHomeDir(), ".config", "cloud-gate", "config.yml"), "An Ini file with credentials")
+	oldBotoCompat        = flag.Bool("oldBotoCompat", false, "add aws_security_token for OLD boto installations (not recommended)")
+)
+
+type AppConfigFile struct {
+	BaseURL              string `yaml:"base_url"`
+	OutputProfilePrefix  string `yaml:"output_profile_prefix"`
+	LowerCaseProfileName bool   `yaml:"lower_case_profile_name"`
+}
+
+type cloudAccountInfo struct {
+	Name           string
+	AvailableRoles []string
+}
+
+type getAccountInfo struct {
+	AuthUsername  string
+	CloudAccounts map[string]cloudAccountInfo
+}
+
+type AWSCredentialsJSON struct {
+	SessionId    string    `json:"sessionId"`
+	SessionKey   string    `json:"sessionKey"`
+	SessionToken string    `json:"sessionToken"`
+	Region       string    `json:"region,omitempty"`
+	Expiration   time.Time `json:"cloudgate_comment_expiration,omitempty"`
+}
+
+func loadVerifyConfigFile(configFilename string) (AppConfigFile, error) {
+	var config AppConfigFile
+	if _, err := os.Stat(configFilename); os.IsNotExist(err) {
+		err = errors.New("No config file: please re-run with -configHost")
+		return config, err
+	}
+	source, err := ioutil.ReadFile(configFilename)
+	if err != nil {
+		err = errors.New("cannot read config file")
+		return config, err
+	}
+	err = yaml.Unmarshal(source, &config)
+	if err != nil {
+		err = errors.New("Cannot parse config file")
+		return config, err
+	}
+
+	if len(config.BaseURL) < 1 {
+		err = errors.New("Invalid Config file... no place get the credentials")
+		return config, err
+	}
+	// TODO: ensure all enpoints are https urls
+	return config, nil
+}
+
+func saveDefaultConfig(configFilename string) error {
```

Comment:
You're replacing the global variable with a local variable with the same name. This will cause confusion.